src/OSPSuite.Core/Domain/Services/IDataRepositoryTask.cs

```diff
@@ -1,72 +1,81 @@
+      ///    If set to <c>true</c> (default), the values will be exported using the display unit of the <see cref="DataColumn" />
+      /// </summary>
+      public bool UseDisplayUnit { get; set; } = true;
+
+      /// <summary>
+      ///    If set to <c>true</c> (default is <c>false</c>), the value will be
+      ///    formatted using the number of decimal  places defined in the application
+      /// </summary>
+      public bool FormatOutput { get; set; } = false;
+
+      /// <summary>
+      ///    If set to <c>true</c> (default is <c>false</c>), the exported columns will have the type set to object instead of
+      ///    float.
       ///    This is required when using the resulting DataTable for direct binding in the UI
       /// </summary>
-      IEnumerable<DataTable> ToDataTable(
-         IEnumerable<DataColumn> dataColumns, 
-         Func<DataColumn, string> columnNameRetriever, 
-         Func<DataColumn, IDimension> dimensionRetriever, 
-         bool formatOutput = false, 
-         bool useDisplayUnit = true,
-         bool forceColumnTypeToObject = false);
+      public bool ForceColumnTypeAsObject { get; set; } = false;
+   }
+
+   public interface IDataRepositoryTask
+   {
+      DataRepository Clone(DataRepository dataRepositoryToClone);
 
       /// <summary>
-      ///    Export the given data repository to excel using the ToDataTable function
+      ///    Returns one datable for each base grid and associated columns representing the content of the repository (one column
+      ///    for each DataColumn)
       /// </summary>
-      void ExportToExcel(IEnumerable<DataColumn> dataColumns, string fileName, bool launchExcel = true);
+      IReadOnlyList<DataTable> ToDataTable(IEnumerable<DataColumn> dataColumns, DataColumnExportOptions exportOptions = null);
 
       /// <summary>
       ///    Export the given data repository to excel using the ToDataTable function
       /// </summary>
-      void ExportToExcel(IEnumerable<DataColumn> dataColumns, string fileName, Func<DataColumn, string> columnNameRetriever, Func<DataColumn, IDimension> dimensionRetriever, bool launchExcel = true);
+      void ExportToExcel(IEnumerable<DataColumn> dataColumns, string fileName, bool launchExcel = true, DataColumnExportOptions exportOptions = null);
 
       /// <summary>
       ///    Export the given data tables to excel
       /// </summary>
       void ExportToExcel(IEnumerable<DataTable> dataTables, string fileName, bool launchExcel = true);
 
       BaseGrid CloneBaseGrid(DataColumn baseGridToClone);
+
       DataColumn CloneColumn(DataColumn sourceColumn, BaseGrid clonedBaseGrid);
+
+      /// <summary>
+      ///    Export the given data repository to excel using the ToDataTable function
+      /// </summary>
+      Task ExportToExcelAsync(IEnumerable<DataColumn> dataColumns, string fileName, bool launchExcel = true, DataColumnExportOptions exportOptions = null);
```

Comment:
I have added the Async methods as well to be consistent with other call we have throughout the suite for IO